packages/component-meta/scripts/generateComponentMeta.ts

```diff
@@ -221,9 +221,31 @@ const generateComponentMeta = (): void => {
     }
 
     let [, rawPropTypes] = /const [a-z][a-zA-z]+: (?:Omit<)?PropTypes<.+?> = ({[\s\S]+?});/.exec(source) || [];
+
+    // handle shared props
+    let sharedPropsExportFile; // also needed for import paths of variables
+    if (rawPropTypes && rawPropTypes.match(/\.{3}[a-zA-Z]+,/g)) {
+      const [, sharedPropsConstName] = rawPropTypes.match(/(shared[a-zA-z]+Types)/);
+      const [, sharedPropsImportPath] = source.match(new RegExp(`${sharedPropsConstName}[\\s\\S]+?from '(.+)';`)) || [];
+      const componentFilePath = componentFiles.find((file) =>
+        file.match(new RegExp(`${tagName.replace(/^p-/, '/')}\\.tsx$`))
+      );
+      const sharedPropsExportFilePath = sharedPropsImportPath.match(/^\./)
+        ? path.resolve(componentFilePath, '..', sharedPropsImportPath) // relative path
+        : sharedPropsImportPath; // absolute path to other package
+
+      sharedPropsExportFile = fs.readFileSync(`${sharedPropsExportFilePath}.ts`, 'utf8');
```

Comment:
Now we are reading this file which could just be reexporting what we are looking for.  
Why not require the constant directly via `require(pathtoFile)[nameOfConstant]` just like down below?